test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py

```diff
@@ -266,6 +269,18 @@ def __init__(self, path, analyze_arg_spec=False, collection=None, base_branch=No
         else:
             self.base_module = None
 
+        if self.collection:
```

Comment:
The routing file should be parsed once per invocation of validate-modules instead of once per module.